### PR TITLE
fix(auth): migrate overview errors to clickhouse

### PR DESCRIPTION
**Pull request by jordienr**
## Problem

Auth overview error tables call the legacy logs endpoint through fetchLogs defaults, even with the ClickHouse migration enabled. Success-rate cards also show zero when there are no requests and misleading relative changes between small rates.

## Fix

Select matching BigQuery or ClickHouse queries with otelLegacyLogs, separate caches by engine, and normalize numeric results.

Show No data for success rates without requests and omit comparisons when either period has no requests. Show success-rate changes in percentage points: 0% to 0.2% displays +0.2 pp. Omit undefined relative changes from a zero baseline for activity and sign-up counts. Show explicit errors for failed log requests, including error payloads returned with HTTP 200, instead of empty tables.

## Validation

- Auth overview error tables compared against staging with matching data.
- 84 focused tests passed across four suites, including 25 direct formatter tests.
- 12 MSW integration tests exercise both endpoint/SQL pairs, HTTP and embedded API failures, and rendered No data, genuine 0%, and +0.2 pp states.
- Unit tests cover missing periods, zero requests, percentage-point and relative changes, SQL structure, and numeric result parsing.
- Formatting and diff checks passed; code review found no actionable issues.
- Full local lint/typecheck are limited by shared checkout dependencies. Browser comparison confirmed the deployed rate display uses percentage points and shows No data without a comparison for absent server requests; populated error rows match staging. The final formatter extraction (9886b78682) was also deployed and verified in the browser; CI completion remains outstanding.

Split from #50173.

## Summary by CodeRabbit

- **New Features**
  - Added OpenTelemetry support for authentication error and metrics reporting.
  - Added clearer handling of missing metric data, including “No data” states.
  - Improved success-rate change calculations using percentage-point differences.
  - Added user-visible error states when analytics requests fail.

- **Bug Fixes**
  - Improved validation and handling of authentication metrics and error data.
  - Corrected formatting and rounding for metric values and changes.

- **Tests**
  - Expanded coverage for legacy and OpenTelemetry analytics, error handling, empty data, formatting, and edge cases.

### CODE REVIEW

File: apps/studio/components/interfaces/Auth/Overview/OverviewMetrics.tsx [hunk 1]
```diff
@@ -49,29 +52,30 @@ import { useSidebarManagerSnapshot } from '@/state/sidebar-manager-state'
...
   const router = useRouter()
-  const formattedCurrent =
-    suffix === 'ms'
-      ? current.toFixed(2)
-      : suffix === '%'
-        ? current.toFixed(1)
-        : Math.round(current).toLocaleString()
-  // const signChar = previous > 0 ? '+' : previous < 0 ? '-' : ''
+  let formattedCurrent = 'No data'
+  if (current !== null) {
```
Comment (jordienr) [inline apps/studio/components/interfaces/Auth/Overview/OverviewMetrics.tsx]: could be extracted to a function and tested

File: apps/studio/components/interfaces/Auth/Overview/OverviewMetrics.test.tsx [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+    expect(requests).toHaveLength(2)
+    for (const request of requests) {
+      expect(request.pathname).toContain('/projects/default/analytics/endpoints/')
+      const start = Date.parse(request.searchParams.get('iso_timestamp_start') ?? '')
+      const end = Date.parse(request.searchParams.get('iso_timestamp_end') ?? '')
+      expect(end - start).toBeCloseTo(24 * 60 * 60 * 1000, -2)
+    }
+  })
+
+  it('keeps both log tables cached separately when the engine flag changes', async () => {
```
Comment (jordienr) [inline apps/studio/components/interfaces/Auth/Overview/OverviewMetrics.test.tsx]: unnecessary tbh

**Review by jordienr — COMMENTED** [review]
nit

**Review by charislam — APPROVED** [review]
